src/moonlink_service/src/lib.rs

```diff
@@ -79,6 +81,10 @@ fn setup_readiness_probe() -> Arc<ServiceStatus> {
 pub async fn start_with_config(config: ServiceConfig) -> Result<()> {
     // Set logging config before service start.
     let _guard = logging::init_logging(config.log_directory.clone());
+    // Set global meter provider config before service start.
+    if config.otel_api_port.is_some() {
```

Comment:
FYI (not request for change): otel api port has nothing to do with meter provider, the former is used for taking otel metrics, the later is used to emit internal metrics.